src/Harald.WebApi/Features/Connections/Infrastructure/DrivingAdapters/Api/ConnectionsController.cs

```diff
@@ -70,11 +82,102 @@ public class ConnectionsController : ControllerBase
                     new {message = validationException.MessageToUser}
                 );
             }
-
-
+            
             var connectionDtos = connections.Select(ConnectionDto.CreateFromConnection);
 
             return Ok(new ItemsEnvelope<ConnectionDto>(connectionDtos));
         }
+
+        [HttpPost]
+        public async Task<IActionResult> AddConnection(ConnectionDto connection)
+        {
+            if (string.IsNullOrEmpty(connection.ClientId))
+            {
+                return BadRequest("ClientId ID is required.");
+            }
+
+            if (string.IsNullOrEmpty(connection.ClientType))
+            {
+                return BadRequest("ClientType is required.");
+            }
+
+            if (string.IsNullOrEmpty(connection.ChannelId))
+            {
+                return BadRequest("Channel ID is required.");
+            }
+
+            if (string.IsNullOrEmpty(connection.ChannelType))
+            {
+                return BadRequest("ChannelType is required.");
+            }
+
+            if (string.IsNullOrEmpty(connection.ChannelName))
+            {
+                return BadRequest("ChannelName is required.");
+            }
+
+            var userGroup = await _slackService.EnsureUserGroupExists(connection.ClientName);
+            var capability = Capability.Create(Guid.Parse(connection.ClientId), connection.ClientName, connection.ChannelId, userGroup.Id);
+
+            await _capabilityRepository.Add(capability);
+
+            var response = await _slackFacade.JoinChannel(connection.ChannelName);
```

Comment:
If the channel does not exist it will be auto created by the Slack API. See more here @ https://api.slack.com/methods/channels.join. I have added logic to box / toString the class for the added validation check. Keep in mind that we get this on the API side as well so we need to decide if we want to keep this logic in two places. :)